solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/CollectionPropertiesZkStateReader.java

```diff
@@ -50,17 +50,14 @@ public class CollectionPropertiesZkStateReader implements Closeable {
    * notified when the properties of the collection change. When a collection's properties change,
    * all registered observers for that collection are notified by a "collectionPropWatcher".
    */
-  private ConcurrentHashMap<String, CollectionWatch<CollectionPropsWatcher>>
+  private final ConcurrentHashMap<String, CollectionWatch<CollectionPropsWatcher>>
       collectionPropsObservers = new ConcurrentHashMap<>();
 
   /** Used to submit notifications to Collection Properties watchers in order */
   private final ExecutorService collectionPropsNotifications =
       ExecutorUtil.newMDCAwareSingleThreadExecutor(
           new SolrNamedThreadFactory("collectionPropsNotifications"));
 
-  private final ExecutorService notifications =
```

Comment:
I realize I was mistaken that it's used for multiple things, so your choice of separating it to a "cache cleaner" one is reasonable.  I like the way you had it in retrospect... but don't care enough either way so do as you like.

Reply:
Reverted to previous commit